soc/arm/silabs_exx32/common/soc.c

```diff
@@ -92,6 +112,14 @@ static int silabs_exx32_init(struct device *arg)
 
 	_ClearFaults();
 
+#ifdef CONFIG_SOC_GECKO_EMU_DCDC
+	/*
+	 * Initialize DCDC before attempting to set clock, HFXO for example
```

Comment:
As @chrta has noticed 'DCDC is not always required for HFXO. It depends on the board/schematic.' I agree with the statement. Let's maybe drop ', HFXO for example requires this' or otherwise we would need to go into too much detail to make this comment viable.